sale_general_discount/models/sale.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from lxml import etree
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    general_discount = fields.Float(
+        digits=dp.get_precision('Discount'),
+        string='Discount (%)',
+    )
+
+    @api.multi
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        for order in self:
+            super(SaleOrder, self).onchange_partner_id()
+            order.general_discount = order.partner_id.sale_discount
+
+    @api.multi
+    @api.onchange('general_discount')
+    def onchange_general_discount(self):
+        with self.env.norecompute():
+            for order in self:
+                order.mapped('order_line').update({
```

Comment:
Doing `self.mapped('order_line').update({` wouldn't be the same? :thinking:

Reply:
That's what I suggested too: https://github.com/OCA/sale-workflow/pull/685#discussion_r212823526